### PR TITLE
[RFC] Fix rpm spec version for nightly tarballs

### PR DESCRIPTION
**Pull request by patrickmacarthur**
This patch series is intended to fix issue #469, with the caveats that I gave in the thread for that issue.  It includes several bugfixes to the commit that I had mentioned in the issue itself.  I have now tested the patch series and it works on my system.

This isn't so much meant as a pull request as a request for comments.  I plan to resubmit with the commits squashed/reordered, but the pull request mechanism should make it easier to give comments.

### CODE REVIEW

File: config/cron-make-nightly-tarball.pl [hunk 1]
```diff
@@ -121,6 +124,31 @@ sub verbose {
 print OUT $config;
 close(OUT);
 
+# git describe output looks like: ${tag}-${count}-g${commitid}
+# Fedora guidelines want RPM release tag to look like:
+# Release: 0.${date}git${commitid}
+my @gdsplit = split(/-/, $gd);
+my $commitid = substr($gdsplit[-1], 1);
```
Comment (jsquyres) [inline config/cron-make-nightly-tarball.pl]: Might be better to use a regexp (untested -- just typed in github web UI, so test this before accepting):

```perl
$gd =~ m/-g(\w+?)$/;
my $commitid = $1;
```

**Comment by jsquyres** [comment]
In principle, I'm ok with this -- it makes a self-contained nightly tarball.

Should we make a nightly SRPM as well?  I don't have any objection to doing that, but I don't know if it's useful or not.  I.e., if we make a nightly SRPM, will anyone use it?

I don't think we -- the community -- want to get in the business of making nightly binary RPMs; there's too many platforms to do this for.  I'm *guessing/assuming* that vendors / testers who care will take the tarball (or SRPM?) and make their own nightly binaries.  Thoughts?

**Comment by patrickmacarthur** [comment]
I would also agree that we should not generate binary RPMs for the same reasons that you gave, unless people really ask for them.  I just wanted to fix the spec file to make it easier for my organization to generate binary RPMs internally (i.e., we can just use the spec file in the tarball as-is).

As for an SRPM, that would be useful to me since it would save a couple steps on my end but I would only go through the effort if others would like to use the SRPM as well.  Maybe we can ask that question at the next call.

Also, FWIW I believe OFED consists of SRPM packages and an installer that builds and installs RPMs from that.

I will test and generate a new pull request incorporating your suggested modification.

**Comment by jsquyres** [comment]
@patrickmacarthur FWIW, in the future, you don't need to close a pull request and open a new one when you want to modify some/all of the commits on the existing PR.  Instead, you can just "git push --force" (e.g., if you rebase/squash/whatever) up to your branch, and github will automatically adjust the existing PR to show your new commits.

This is just about the only time it's socially acceptable to git push force to a public repo, though.  :-)

**Pull request closed without merging**